crates/sui-types/src/sui_system_state/sui_system_state_inner_v1.rs

```diff
@@ -122,6 +129,25 @@ impl ValidatorMetadataV1 {
             )),
         }?;
 
+        let next_epoch_pop = match self.next_epoch_proof_of_possession.clone() {
+            None => Ok::<Option<narwhal_crypto::Signature>, u64>(None),
+            Some(bytes) => Ok(Some(
+                narwhal_crypto::Signature::from_bytes(bytes.as_ref())
+                    .map_err(|_| E_METADATA_INVALID_POP)?,
+            )),
+        }?;
+        // Verify proof of possession for the next epoch protocol key
+        if let Some(next_epoch_pop) = next_epoch_pop {
+            if let Some(ref next_epoch_protocol_pubkey) = next_epoch_protocol_pubkey {
+                verify_proof_of_possession(
+                    &next_epoch_pop,
+                    next_epoch_protocol_pubkey,
+                    self.sui_address,
+                )
+                .map_err(|_| E_METADATA_INVALID_POP)?;
+            }
+        }
```

Comment:
we should enforce here that "if next_epoch_protocol_pubkey is not none, then next_epoch_pop should be present and attests to the pub key"

Reply:
added check. although idt there is any entry function that allow this, either create both at none or update both to some. iiuc i cannot create a raw validator_metadata in move bypassing the entry functions.